### PR TITLE
DD-412 - Imported descriptions should preserve newlines using html elements

### PR DESCRIPTION
**Pull request by ChrisMuller80**
Fixes DD-412 

Note: @jo-pol is looking at the preserving of whitespaces in EASY. Currently easy-deposit-form removes all spacing.

# Description of changes

# How to test
1. Change  dataset.xml description field of an example bag to include a newline (with enter) and paragraph (enter twice)
2. Import the bag `dd-dans-deposit-to-dataverse import -s <path to bag> <outboxdir>`
3. Check dataset in UI
4. Also check unit tests

# Related PRs 
(Add links)
* 

# Notify
@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala [hunk 1]
```diff
@@ -42,4 +42,11 @@ object Description extends BlockCitation {
     m.addPrimitiveField(DESCRIPTION_VALUE, s"$prefix: ${ node.text }")
     m.toJsonObject
   }
+
+  def newlineToHtml(description: String): String = {
+    description.split("\n{2}")
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala:47]: * You will get an empty paragraph if the description starts with two new lines.
* Can the description somehow originate from windows and thus also have `\r` next to `\n`?

File: src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala [hunk 2]
```diff
@@ -42,4 +42,16 @@ object Description extends BlockCitation {
...
+
+  def newlineToHtml(description: String): String = {
+    //OS agnostic newline and paragraph regex
+    val newline = "\r\n|\n|\r"
+    val paragraph = "(\r\n){2,}|\n{2,}|\r{2,}"
+    description
+      .trim
+      .split(paragraph)
+      .map(p => s"<p>$p</p>")
+      .map(_.replaceAll(newline, "<br>"))
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala:54]: @ChrisMuller80 : does EASY also preserve single newlines like this? I thought it only converted blank lines to paragraphs. Please verify.

**Comment by janvanmansum** [comment]
@ChrisMuller80 : merging this, please check about the `<br>` elements (see question above).

File: src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala (hunk 2 above)
Comment (ChrisMuller80) [reply src/main/scala/nl.knaw.dans.easy.dd2d/mapping/Description.scala:54]: @janvanmansum : I checked by looking at the page source of datasets in easy.
e.g.
1. for newlines see: view-source:https://easy.dans.knaw.nl/ui/datasets/id/easy-dataset:206964
2. for paragraph see: view-source:https://easy.dans.knaw.nl/ui/datasets/id/easy-dataset:131490
 
I created my own implementation. 
Note: more than two newlines are reduced to a single paragraph.

**Pull request merged**